src/showroom_connector/rq.py

```diff
@@ -0,0 +1,23 @@
+from datetime import timedelta
+
+from django_rq import get_queue
+from rq.exceptions import NoSuchJobError
+from rq.registry import ScheduledJobRegistry
+
+from django.conf import settings
+
+
+def enqueue(job_id, function, **kwargs):
+    queue = get_queue('default')
+    registry = ScheduledJobRegistry(queue=queue)
+    if job_id in registry.get_job_ids():
+        try:
+            registry.remove(job_id, delete_job=True)
+        except NoSuchJobError:
+            pass
+    queue.enqueue_in(
+        timedelta(seconds=settings.WORKER_DELAY),
```

Comment:
since there's no comment, just a quick question to understand, why is a delay used here?

Reply:
Not totally sure, but I think we needed to introduce a short delay due to some concurrency issues. Will try to figure out more tomorrow and either add a comment or drop the delay.

All other review comments are resolved through the 4 commits below.

Reply:
I have annotated the enqueue functions now, as it is a wrapper to deal with concurrency and quick consecutive updates of user preferences.